js/app.js

```diff
@@ -142,10 +144,36 @@ function appendBlogTagToTagCloud(blogTag) {
 
 
 function navigateToBlogs(blogTag) {
-    const baseURL = location.href;
-    const url = new URL(location.pathname, baseURL);
-    if (blogTag) { url.searchParams.set("tag", blogTag); }
-    location.href = url.toString();
+    var toggledTags = toggledBlogTags(blogTag);
+    if (toggledTags) baseUrl.searchParams.set("tags", toggledTags);
+    window.location.href = baseUrl.href;
+}
+
+
+function toggledBlogTags(blogTag) {
+    if (tags === undefined) { return blogTag; }
+    var selectedTags = toggleTagSelection(tags, blogTag);
+    return selectedTags;
+}
+
+
+function toggleTagSelection(selectedTags, blogTag) {
+    var arrayTags = selectedTags.split(",");
+    if (arrayTags.includes(blogTag)) {
+        arrayTags = arrayTags.filter(function (value) {
+            return value !== blogTag;
+        });
+    } else {
+        arrayTags.push(blogTag);
+    }
+    return arrayTags.join(",");
+}
+
```

Comment:
this tag toggling thing might make a good candidate for breaking out into a separate module. You could even use a class if you like.

Also, consider using a Set instead of this string munging logic (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set#Examples)

Reply:
Not sure what you mean by string munging logic. Tags are passed as string, you'd still need to split and join

Reply:
There are a couple of options here:
1. You could store tags as a Set (up on line 6) and just convert it back to a string when you are ready to navigate
2. Instead of storing tags as a CSV, you could use the natural querystring method for multi-valued parameters: https://codepen.io/goofballLogic/pen/YzXWpEy?tag=hello&tag=world&tag=other

Reply:
> this tag toggling thing might make a good candidate for breaking out into a separate module. You could even use a class if you like.
> 

I am interested in this, but need more research so in another future PR...